DSpace/dspace-angular
# Edit collection item template

## References
This PR closes [#662](https://github.com/DSpace/dspace-angular/issues/662)

## Description
This PR adds the ability to edit the template item of collections.
This PR is something we created a while back, but was never completely finished due to the reduced priority of the item template feature. However, this PR fixes issue [#662](https://github.com/DSpace/dspace-angular/issues/662), so we'd like to contribute this now.

## Instructions for Reviewers
Changes made:
- Added `ItemTemplateDataService`
- Added `EditItemTemplatePageComponent` at route `/collection/${collection-uuid}/itemtemplate`. This page displays the metadata of the collection's template item and the ability to edit them (similar to edit metadata of a regular item, it uses the same child component)
- Added template item section on top of the edit metadata page of a collection. When no template item is present yet, an "Add" button is displayed. Clicking this button will create a template item and redirect the user to the edit page. When a template item is present, an "Edit" and "Delete" button will be displayed.

How to test:
- Go to a collection's edit page
- Create a new template item and edit some of its metadata
- Go back to the collection's edit page, it should now show an "Edit" and "Delete" button
- Clicking "Edit" should bring you back to the edit metadata page of the template item
- Clicking "Delete" should delete the template item
- Starting a new submission in a collection that contains a template item should fill in the metadata from the template item automatically into the new submission. But that doesn't work at the moment: #748. That's unrelated to this PR

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
    * Include tests for different user types (if behavior differs), including: (1) Anonymous user, (2) Logged in user (non-admin), and (3) Administrator.
    * Include tests for error scenarios, e.g. when errors/warnings should appear (or buttons should be disabled).
    * For bug fixes, include a test that reproduces the bug and proves it is fixed. For clarity, it may be useful to provide the test in a separate commit from the bug fix.
